### DISCUSSION TITLE
Axis Distribution misorientation

### DISCUSSION THREAD

**Discussion by antonthiel** [mtex-toolbox/mtex#982] [Ask Anything] [answered]
Good evening, 
is it possible to plot the axis distrubtion of the gb misorientation for a specific degree interval on the IPF?
For example for the distribution of the gb misorienation  between 20 and 40 degree. 
i've got the gb.misorientations for the whole spectrum of angles
Thanks a lot!

**Comment by kilir** [comment]
Hi,
how about:
```
mtexdata titanium
grains = ebsd.calcGrains('angle',10*degree)

gb=grains.boundary('indexed')
cond  =  gb.misorientation.angle > 20*degree & gb.misorientation.angle < 40*degree;
plotAxisDistribution(gb.misorientation(cond))

```
Cheers,
Rüdiger

**Reply by antonthiel** [reply]
Dear Rüdiger, 
unfortunately your suggestion does not work. 
I have searched for a long time in the script but have not found where to put a condition for the angles.
I have also attached the properties of the variable :
![property](https://user-images.githubusercontent.com/81169124/115843535-38463a00-a41f-11eb-9e4f-626c85bd1cf9.PNG)

The following text is the script of the command plotAxisDistribution:

function h = plotAxisDistribution(obj,varargin)
% plot axis distribution
%
% Syntax
%
%   plotAxisDistribution(cs)        % random axis distribution
%   plotAxisDistribution(cs1,cs2)   % random misorientation axis distribution
%   plotAxisDistribution(mori)      % axes in crystal coordinates
%   plotAxisDistribution(mori,cs1)  % axes in crystal coordinates
%   plotAxisDistribution(ori1,ori2) % axes in specimen coordinates
%   plotAxisDistribution(odf)
%
% Input
%  odf - @ODF
%  mori - @misorientation
%  ori1,ori2 - @orientation
%  cs1, cs2 - @crystalSymmetry
%
% Options
%  RESOLUTION - resolution of the plots
%
% Flags
%  antipodal - include <VectorsAxes.html antipodal symmetry>
%  complete  - plot entire (hemi)--sphere
%  noLabel   - do not annotate
%
% See also
% S2Grid/plot savefigure Plotting Annotations_demo ColorCoding_demo PlotTypes_demo
% SphericalProjection_demo

[mtexFig,isNew] = newMtexFigure(varargin{:});

res = 2.5 * degree;
if isa(obj,'symmetry')
  res = .5 * degree;
  cs1 = obj;
  if nargin > 1 && isa(varargin{1},'symmetry')
    cs2 = varargin{1};
  else
    cs2 = specimenSymmetry;
  end
elseif isa(obj,'orientation')

  obj = axis(obj,varargin{:});

else
  cs1 = obj.CS;
  cs2 = obj.SS;
end


if isa(obj,'vector3d')

  h = plot(obj,'symmetrised','FundamentalRegion',varargin{:});

else

  if isa(obj,'orientation') && isa(varargin{1},'crystalSymmetry')
    dcs = varargin{1};
  else
    dcs = calcDisjoint(cs1,cs2,varargin{:});
  end

  % plotting grid
  sR = fundamentalSector(dcs,varargin{:});
  h = plotS2Grid(sR,'resolution',res,varargin{:});

  % plot
  varargin = [delete_option(varargin,'complete'),dcs.plotOptions];
  if isa(obj,'symmetry')
    density = pos(calcAxisDistribution(cs1,cs2,h,varargin{:}));
  else
    density = pos(calcAxisDistribution(obj,h,varargin{:}));
  end
  h = smooth(h,density,dcs,varargin{:});

end

if isNew % finalize plot
  name = inputname(1);
  set(gcf,'Name',['Axis Distribution of ',name]);
  mtexFig.drawNow('figSize',getMTEXpref('figSize'),varargin{:});
end

if nargout == 0, clear h; end

  function dcs = calcDisjoint(cs1,cs2,varargin)

    dcs = disjoint(cs1.properGroup,cs2.properGroup);
    if check_option(varargin,'antipodal') || ...
        ((isa(obj,'orientation') || isa(obj,'ODF')) && obj.antipodal)
      dcs = dcs.Laue;
    end
  end

end

function d = pos(d)

if min(d(:)) > -1e-5, d = max(d,0);end

end

**Reply by antonthiel** [reply]
this is the script for the commant plotAngleDistrubition. 
could not the two be combined here to solve the problem?

function h = plotAngleDistribution(obj,varargin)
% plot axis distribution
%
% Syntax
%
%   plotAngleDistribution(mdf)
%   plotAngleDistribution(CS1,CS2)
%   plotAngleDistribution(grains.boundary.misorientation)
%
% Input
%  CS - @symmetry
%
% Options
%  resolution, xy*degree - resolution of the plots (given as angle)
%

[mtexFig,isNew] = newMtexFigure(varargin{:}); 
mtexFig.keepAspectRatio = false;


% compute angles
plotType = 'line';
if isa(obj,'symmetry')
  maxOmega = maxAngle(obj,varargin{:});
else
  maxOmega = maxAngle(obj.CS,obj.SS);
  if ~isa(obj,'ODF'), plotType = 'bar'; end
end

% seach for existing bar plots and adjust bar center
h = findobj(mtexFig.gca,'type','bar','-or','type','hgGroup');
h = flipud(h(:));

unit = '%';
if ~isempty(h)

  midPoints = ensurecell(get(h,'XData'));
  midPoints= midPoints{1}*degree;
  bins = [0 10 20 30 40 50];
  bins = (bins(1:end-1) + bins(2:end))/2;
  density = ensurecell(get(h,'YData'));
  density = cellfun(@(x) x(:),density,'UniformOutput',false);
  density = horzcat(density{:});
  lg = ensurecell(get(h,'DisplayName'));

  if strcmp(plotType,'bar')
    delete(h); % remove old bars
  
    % add a new column
    density(:,end+1) = 0;
    
    % maybe we have to enlarge bins
    if maxOmega > max(bins)
      bins = 0:(bins(2)-bins(1)):maxOmega + 0.01;
      density(end+1:length(bins)-1,:) = 0;
    end
  else
    faktor = 100 / size(density,1);
  end
  
else
  
  if strcmp(plotType,'bar')
  
    % bin size given?
    if max(obj.angle) < maxOmega/2, maxOmega = max(obj.angle);end
    nbins = max(15,round(maxOmega/get_option(varargin,'resolution',5*degree)));
    
    % compute bins
    bins = linspace(-eps,maxOmega+0.01,nbins);
    density = zeros(nbins-1,1);
    lg = {};
  elseif check_option(varargin,'percent')
    faktor = 100;    
  else
    faktor = 1;
    unit = 'mrd';
  end
end


% compute angle distribution
if isa(obj,'symmetry') || isa(obj,'ODF')
  [density,omega] = calcAngleDistribution(obj,varargin{:});
else  
  d = histcounts(obj.angle,bins).';
  midPoints = 0.5*(bins(1:end-1) + bins(2:end));
  density(:,end) = 100 * d ./ sum(d);  
end

% plot angle distribution
if strcmp(plotType,'bar')

  h = optiondraw(bar(midPoints/degree,density,'parent',mtexFig.gca),varargin{:});
  xlim(mtexFig.gca,[0,max(bins)/degree])

  % update legend
  lg = [lg;{[obj.CS.mineral '-' obj.SS.mineral]}];
  for i=1:length(h)
    set(h(i),'DisplayName',lg{i});
  end

else

  h = optiondraw(plot(omega/degree,faktor * max(0,density),...
    'parent',mtexFig.gca),'LineWidth',2,varargin{:});

end
  
% finish  
if isNew
  xlabel(mtexFig.gca,'Misorientation angle (degrees)');
  ylabel(mtexFig.gca,['Frequency (' unit ')']);
  drawNow(mtexFig,varargin{:});
end

if nargout == 0, clear h; end

**Reply by kilir** [reply]
You may have to specify what you mean by does not work" ? obviously it should be `grains.boundary().misorientation` not `grains.boundary().misorientations`
Cheers,
Rüdiger

**Reply by antonthiel** [reply]
now it works! thanks a lot rüdiger! :)

**Comment by antonthiel** [comment, marked as answer]
if someone has the same question again, here is the correct code:_
```
grains = calcGrains(ebsd)
gb=grains.boundary('indexed')
cond  =  gb.misorientation.angle > Y*degree & gb.misorientation.angle < X*degree;
plotAxisDistribution(gb.misorientation(cond), 'contour')
```
from https://github.com/mtex-toolbox/mtex/discussions/982#discussioncomment-648469
